client/src/components/v3/dashboard/V3SpeedDistanceChart.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import { Sensor, useSensorData } from 'api/common/data';
+import { useChannel } from 'api/common/socket';
+import SpeedDistanceChart from 'components/common/charts/SpeedDistanceChart';
+import { ChartPoint } from 'types/chart';
+import { ReedDistanceRT, ReedVelocityRT } from 'types/data';
+
+export const V3SDChartKey = 'v3-dashboard-speed-distance-chart-data';
+
+/**
+ * Passes V3 data to the speed distance chart component
+ *
+ * @returns Component
+ */
+export function V3SpeedDistanceChart() {
+  const storedData = sessionStorage.getItem(V3SDChartKey);
+
+  // Used to store the data points
+  const [data, setStateData] = useState<ChartPoint[]>(
+    storedData ? JSON.parse(storedData) : [],
+  );
+  const maxPoint = useRef(-1);
+
+  // Reset when start message received
+  const reset = () => setStateData([]);
+  useChannel('module-3-start', reset);
+
+  // Store data for session
+  const setData = (newData: ChartPoint[]) => {
+    sessionStorage.setItem(V3SDChartKey, JSON.stringify(newData));
+    setStateData(newData);
+  };
+
+  // Speed
+  const point = useSensorData(3, Sensor.ReedVelocity, ReedVelocityRT);
```

Comment:
Another thing I only just realised, and I apologise - Can we use GPS speed instead of reed here? At high speeds, GPS speed is more accurate as there's no uncertainty in wheel radius (which is a function of the tyre we used, the tyre pressure, etc)

Ideally we'd use GPS but then fall back to reed if GPS is unavailable, but don't worry about this for now. Something to worry about later :)

Also, unsure what the name `point` means - `speed` or `velocity`?

Reply:
`point` is meant to be a point on the graph and I named it that way because I didn't know whether `ReedVelocity` was meant to be speed or velocity.
Changed to GPS speed and renamed `point` to `speed`.
ed02064